wizeline/remix-project-lab
# Create profile from datalake on login

# What does this PR do?

If when login in, there is no profile with the same email than the auth0 gmail email you used on our database, then it will query the datalake to get the profile information.

You need a new `GOOGLE_APPLICATION_CREDENTIALS="/Users/joaquin.bravo/Downloads/[your-key-file].json"` variable on your `.env` file.

This file is on our keybase folder for `projectlab`.

# Where should the reviewer start?

- `app/lake.server.tsx` to see how we are querying the data lake
- `app/auth.server.ts` to see how we changed the authentication code to query the datalake when there is no profile on our own database, and if we find a profile with the same email on the datalake, then we use the info from the datalake to create a new profile on our own database.

# How should this be manually tested?

1. Sign out of your localhost session
2. Delete your email entries from the "User" and "Profiles" table
3. Sign in, and see how the both tables get populated, the Profiles table with data from the datalake.

# Any background context you want to provide?

We need to follow up with a new ticket to change the "Profiles" table to add this columns as simple string columns instead of separate tables to facilitate creating profiles:

- contact__preferred_name
- contact__employee_number (bamboo number)
- contact__location
- contact__country
- contact__status
- avatarUrl (this data is not on the datalake at this moment)

Status can be any of these values:

- NULL
- Bench
- Assigned
- Terminated
- Active Hold
- Proposed
- Assignment to start

I already requested adding the `avatarUrl` column to the datalake.

# What are the relevant tickets?

#65 

# Screenshots

# Questions

# Checklist
